MoltenVK/MoltenVK/Commands/MVKCmdAccelerationStructure.mm

```diff
@@ -0,0 +1,207 @@
+    {
+        [accStructEncoder
+         copyAndCompactAccelerationStructure:_srcAccelerationStructure
+         toAccelerationStructure:_dstAccelerationStructure];
+        
+        return;
+    }
+    
+    [accStructEncoder
+         copyAccelerationStructure:_srcAccelerationStructure
+         toAccelerationStructure:_dstAccelerationStructure];
+}
+
+#pragma mark -
+#pragma mark MVKCmdCopyAccelerationStructureToMemory
+
+VkResult MVKCmdCopyAccelerationStructureToMemory::setContent(MVKCommandBuffer*                  cmdBuff,
+                                                             VkAccelerationStructureKHR         srcAccelerationStructure,
+                                                             uint64_t                           dstAddress,
+                                                             VkCopyAccelerationStructureModeKHR copyMode) {
+    
+    MVKAccelerationStructure* mvkSrcAccStruct = (MVKAccelerationStructure*)srcAccelerationStructure;
+    _mvkDevice = mvkSrcAccStruct->getDevice();
+    _dstAddress = dstAddress;
+    _copyMode = copyMode;
+    
+    MVKAccelerationStructure* mvkDstAccStruct = (MVKAccelerationStructure*)_mvkDevice->getAccelerationStructureAtAddress(_dstAddress);
+    
+    _srcAccelerationStructure = mvkSrcAccStruct->getMTLAccelerationStructure();
+    _dstAccelerationStructure = mvkDstAccStruct->getMTLAccelerationStructure();
+    return VK_SUCCESS;
+}
+                                        
+void MVKCmdCopyAccelerationStructureToMemory::encode(MVKCommandEncoder* cmdEncoder) {
+    id<MTLAccelerationStructureCommandEncoder> accStructEncoder = cmdEncoder->getMTLAccelerationStructureEncoder(kMVKCommandUseNone);
+    
+    if(_copyMode == VK_COPY_ACCELERATION_STRUCTURE_MODE_COMPACT_KHR)
+    {
+        [accStructEncoder
+         copyAndCompactAccelerationStructure:_srcAccelerationStructure
+         toAccelerationStructure:_dstAccelerationStructure];
+        
+        return;
+    }
+    
+    [accStructEncoder
+         copyAccelerationStructure:_srcAccelerationStructure
+         toAccelerationStructure:_dstAccelerationStructure];
+}
+
+#pragma mark -
+#pragma mark MVKCmdCopyMemoryToAccelerationStructure
+
+VkResult MVKCmdCopyMemoryToAccelerationStructure::setContent(MVKCommandBuffer* cmdBuff,
+                                                                uint64_t srcAddress,
+                                                                VkAccelerationStructureKHR dstAccelerationStructure,
+                                                                VkCopyAccelerationStructureModeKHR copyMode) {
+    MVKAccelerationStructure* mvkDstAccStruct = (MVKAccelerationStructure*)dstAccelerationStructure;
+    _mvkDevice = mvkDstAccStruct->getDevice();
+    _srcAddress = srcAddress;
+    _copyMode = copyMode;
+    
+    MVKAccelerationStructure* mvkSrcAccStruct = (MVKAccelerationStructure*)_mvkDevice->getAccelerationStructureAtAddress(_srcAddress);
```

Comment:
And *vice versa* for `vkCmdCopyMemoryToAccelerationStructure()`. Note that here, the `mode` must be `VK_COPY_ACCELERATION_STRUCTURE_MODE_DESERIALIZE_KHR`.

Reply:
@cdavis5e I've resolved this